tests/integration_tests/build/test_coverage.py

```diff
@@ -111,29 +86,20 @@ def test_coverage(monkeypatch, record_property, metrics):
 
     utils.run_cmd(cmd)
 
-    # Extract coverage from html report.
-    #
-    # The line looks like `<abbr title="44724 / 49237">90.83 %</abbr></p>` and is the first
-    # occurrence of the `<abbr>` element in the file.
-    #
-    # When we update grcov to 0.8.* we can update this to pull the coverage from a generated .json
-    # file.
-    index = open(
-        f"./build/cargo_target/{TARGET}/debug/coverage/index.html", encoding="utf-8"
-    )
-    index_contents = index.read()
-    end = index_contents.find(" %</abbr></p>")
-    start = index_contents[:end].rfind(">")
-    coverage_str = index_contents[start + 1 : end]
-    coverage = float(coverage_str)
-
-    # Record coverage.
-    record_property(
-        "coverage", f"{coverage}% {coverage_target}% ±{COVERAGE_MAX_DELTA:.2f}%"
-    )
-    metrics.set_dimensions({"cpu_arch": ARCH})
-    metrics.put_metric("code_coverage", coverage, unit="Percent")
+    # Only upload if token is present and we're in EC2
+    if "CODECOV_TOKEN" in os.environ and global_props.instance:
+        pr_number = os.environ.get("BUILDKITE_PULL_REQUEST")
+        _, branch, _ = utils.run_cmd("git rev-parse --abbrev-ref HEAD")
```

Comment:
nit: you can do `if ... and global_props.is_ec2:`

Reply:
Done!